### PR TITLE
(4/4) Use Libdrop storage

### PR DESCRIPTION
**Pull request by EmilijusS**

### CODE REVIEW

File: fileshare/event_manager.go [hunk 1]
```diff
@@ -301,148 +252,156 @@ func (em *EventManager) handleTransferFinishedEvent(eventJSON json.RawMessage) {
...
+	storageTransfer, ok := storageTransfers[id]
 	if !ok {
 		return nil, ErrTransferNotFound
 	}
+	return storageTransfer, nil
+}
 
-	transferCloned, ok := proto.Clone(transfer).(*pb.Transfer)
+// Storage doesn't contain momentary info about transfer progress, so update it from liveTransfers
+func updateTransferWithLiveData(transfer *pb.Transfer, liveTransfers map[string]*LiveTransfer) {
```
Comment (bartoszWojciechO) [inline fileshare/event_manager.go:389]: Doc string should start with function name.

Comment (bartoszWojciechO) [inline fileshare/event_manager.go:389]: I'd change this function to return `*pb.Transfer`. It's not necessary  but assignment is more verbose/explicit.

File: fileshare/storage/combined.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	jsonTransfers, err := c.json.Load()
+	if err == nil {
+		for key, value := range jsonTransfers {
+			libdropTransfers[key] = value
+		}
+	} else if !errors.Is(err, os.ErrNotExist) {
```
Comment (bartoszWojciechO) [inline fileshare/storage/combined.go:36]: Let's also add logging for other errors.

**Review by bartoszWojciechO — APPROVED** [review]

File: fileshare/events.go [hunk 3]
```diff
@@ -38,16 +39,17 @@ type requestReceivedEvent struct {
 	Files      []*pb.File
 }
 
-type requestQueuedEvent struct {
-	TransferID string `json:"transfer"`
-	Peer       string
-	Files      []*pb.File
-}
+// These are unused at the moment
```
Comment (Savolro) [inline fileshare/events.go:42]: Commented code can be removed

**Review by Savolro — CHANGES_REQUESTED** [review]

**Review by Savolro — APPROVED** [review]

**Pull request merged**